### PR TITLE
[polaris.shopify.com] Add new `/design/icons` design guidance

### LINKED ISSUE #10694: [Markdow] transfer Icons pages
[Doc](https://docs.google.com/document/d/13LrzMI5SMzDZ3Wm117Yz4olE7PzBCFQh2_B5C07WQlE/edit#heading=h.6hnayl6le7qe)

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Fixes #10694  

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/content/design/icons/creating-icons.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
+---
+title: Creating Icons
+description: Icons enhance an experience by providing intuitive and efficient navigation, conveying information concisely, and making it more visually appealing.
```
Comment (yurm04) [inline polaris.shopify.com/content/design/icons/creating-icons.md:3]: @zecarlostorre @tjonx are we using ledes in the L3 pages? if so we'll need un updated lede/description here.

Also need the keywords for this page

File: polaris.shopify.com/content/design/icons/creating-icons.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        Use filled icons for semi-permanent selected states, like pinning an app.
+
+      </Do>
+
+    </Grid.Cell>
+
+    <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 6, lg: 6, xl: 6}}>
+
+      <Dont>
+        ![](/images/design/icons/icons-overview-universal-1@2x.png)
```
Comment (yurm04) [inline polaris.shopify.com/content/design/icons/creating-icons.md]: need the rest of the assets for this `/design/icons` guidance, just been reusing the ones from overview

File: polaris.shopify.com/src/components/Markdown/components/DirectiveCard/index.tsx [hunk 3]
```diff
@@ -22,7 +22,7 @@ export enum DirectiveStatusName {
 
 export type DirectiveStatus = DirectiveStatusName;
 interface DirectiveCardProps extends CardProps {
-  status: DirectiveStatus;
+  status?: DirectiveStatus;
```
Comment (yurm04) [inline polaris.shopify.com/src/components/Markdown/components/DirectiveCard/index.tsx:25]: Sometimes these cards should not have a pill indicating status (just a plain card)

File: polaris.shopify.com/content/design/icons/index.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    </Grid.Cell>
+
+    {/* Dont */}
+    <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 6, lg: 4, xl: 4}}>
+
+      <Dont>
+
+        ![](/images/design/icons/icons-overview-clear-2@2x.png)
+
+        Draw simple objects that are easy to identify.
```
Comment (zecarlostorre) [inline polaris.shopify.com/content/design/icons/index.md]: Replace this string with:
Embellish the icon with unnecessary details.

File: polaris.shopify.com/content/design/icons/creating-icons.md [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+<Lede>{frontmatter.description}</Lede>
+
+<Subnav />
+
+<Stack gap='4'>
+<Card>
+  ## Icon layout
+
+  <InlineGrid gap="4" columns="2">
+    <DirectiveCard status="Tip">
```
Comment (zecarlostorre) [inline polaris.shopify.com/content/design/icons/creating-icons.md]: Can you remove the "tip" badge from the first 4 examples in this page?

File: polaris.shopify.com/content/design/icons/creating-icons.md [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+
+    </Do>
+
+  </Grid.Cell>
+  <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 3, lg: 6, xl: 6}}>
+    <Dont>
+    
+      ![](/images/design/icons/icons-creating-rules-2@2x.png)
+    
+      Icons are two-dimensional, and objects face forward.
```
Comment (zecarlostorre) [inline polaris.shopify.com/content/design/icons/creating-icons.md]: Replace this string with:
Use perspective and 3D objects.

File: polaris.shopify.com/content/design/icons/creating-icons.md [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+
+    </Grid.Cell>
+
+    <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 3, lg: 6, xl: 6}}>
+
+      <DirectiveCard>
+
+        ![](/images/design/icons/icons-creating-filled-2@2x.png)
+
+        Filled icons should align closely to their outline pairs. They don’t need to be a pixel perfect match, but the transition from one to the other must feel smooth.
```
Comment (zecarlostorre) [inline polaris.shopify.com/content/design/icons/creating-icons.md]: Replace this string with:
Icons are made out of contiguous shapes. Ideally one, but it’s possible to go up to three, when required for clarity.

File: polaris.shopify.com/content/design/icons/using-icons.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+            Use icons consistently once a meaning is assigned to them.
+
+          </Do>
+        </Grid.Cell>
+
+        <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 6, lg: 12, xl: 12}}>
+          <Dont>
+            ![](/images/design/icons/icons-using-6@2x.png)
+
+            Use icons consistently once a meaning is assigned to them.
```
Comment (zecarlostorre) [inline polaris.shopify.com/content/design/icons/using-icons.md]: Replace this string with:
Overuse icons. Redundant icons lead to visual clutter and increased cognitive load.

File: polaris.shopify.com/content/design/icons/using-icons.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+      Icon size and weight was specifically crafted to pair with typography.
+
+    </Grid.Cell>
+
+    <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 6, lg: 4, xl: 4}}>
+
+      <Do>
+        ![](/images/design/icons/icons-using-7@2x.png)
+
+        Use icons to replace text for common actions with universally understood icons.
```
Comment (zecarlostorre) [inline polaris.shopify.com/content/design/icons/using-icons.md]: Replace this string with:
Align icons vertically with accompanying typography.

File: polaris.shopify.com/content/design/icons/using-icons.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+      </Do>
+
+    </Grid.Cell>
+
+    <Grid.Cell columnSpan={{xs: 6, sm: 6, md: 6, lg: 4, xl: 4}}>
+
+      <Dont>
+        ![](/images/design/icons/icons-using-8@2x.png)
+
+        Use ambiguous icons without a text label, as that can lead to confusion and misinterpretation.
```
Comment (zecarlostorre) [inline polaris.shopify.com/content/design/icons/using-icons.md]: Replace this string with:
Resize icons. Resizing them will distort the relationship that was established with typography.

**Review by zecarlostorre — CHANGES_REQUESTED** [review]
Thank you Yuraima, apologies for the delay.
I left comments in line where I found strings with the wrong text.

Besides that:
1. The icons missing here are:
EditMinor - Creating icons
PaintBrushMajor - Using icons
Screenshot 2023-09-28 at 18 51 03

2. Most of my images show up cropped because they're 4:3, can we adjust that? Screenshot 2023-09-28 at 18 51 15

3. These are the wrong visuals. The visuals in this section are numbered, and the name is:
icons-creating-filled-vs-outline-...
Screenshot 2023-09-28 at 18 51 27

4. Can we force these to align?
Screenshot 2023-09-28 at 18 54 08

So they look like this instead:
Screenshot 2023-09-28 at 19 11 39

File: polaris.shopify.com/content/design/icons/creating-icons.md (hunk 1 above)
Comment (zecarlostorre) [reply polaris.shopify.com/content/design/icons/creating-icons.md:3]: Keywords for this sections could be:
- Icons
- Icon Design
- User Interface Design
- Visual Navigation
- Consistent Icon Style
- Universally Recognized Symbols
- Icon Creation
- Icon Layout
- 2D Icons
- Icon Stroke Weight
- Filled Icons
- Outlined Icons
- Icon Navigation
- Iconography
- Icon Visual Weight
- Icon Design Guidelines
- Icon Use in Interface
- Icon and Typography
- Icon Size and Weight
- Icon Clarity
- Icon Meaning
- Icon Aesthetics
- Icon Functionality
- Icon Grid Alignment
- Icon Visual Harmony.

Comment (yurm04) [reply polaris.shopify.com/content/design/icons/creating-icons.md:3]: thanks @zecarlostorre ! any updates for the description for this page?

Any updates for the Using Icons page?

**Comment by yurm04** [comment]
Hey @zecarlostorre 👋 

> Most of my images show up cropped because they're 4:3, can we adjust that?

> Can we force these to align?

Mind making two issues for these and add them to the backlog? These would need to be changes to the component that will affect all of the cards so we should fix separate to this PR

**Pull request merged**